### ISSUE TITLE
Journal notification email parser not finding some concepts

### ISSUE THREAD

**Issue by debrarian** [CDL-Dryad/dryad-product-roadmap#637]
**Bug description:**
Processing email notifications from some journals (Development, Journal of Cell Science, Palaeontology, Ecological Applications) is now always failing. The webapp (which can be found in "dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadEmailSubmission.java")

The issue may be related to the existence of duplicate concepts. I noticed that at least some of the journals with failing emails had recent automatically created concepts, resulting in duplicate concepts. This should not happen. In the case of duplicates, I tried removing one of the duplicates through the interface, but the parser still fails.

**Steps to reproduce:**
1. Open the "journal-submit-app" email account and open the "journal-submit-errors" tab.
2. For each of the email notifications listed in the "journal-submit-errors" tab, change the tags to (1) add "journal-submit" tag and (2) remove the "journal-submit-errors" tag.
3. As a result, emails should move from the "journal-submit-errors" tab to the "journal-submit" tab, where they will wait for processing, which occurs automatically about every 5 minutes. 

**Notes:** 

Emails can be retrieved and processed manually by entering the following in a browser:
`https://v1.datadryad.org/journal-submit/retrieve`

To test to see if the webapp is running or to clear any processes that are "stuck", type the following in a browser:
`https://v1.datadryad.org/journal-submit/test`
`https://v1.datadryad.org/journal-submit/clear`

Track the email processing webapp by watching the journal-submit.log with the following command: 
`tail -f /opt/dryad/log/journal-submit.log`

**Expected behavior: **
The webapp (which can be found at "dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadEmailSubmission.java") should eventually parse any emails in the "journal-submit" tab and move them into the appropriate folder for the journal.

**Comment by ryscher** [comment]
- Development -- Current messages are being processed. There were a lot of errors until Feb 10th, when they stopped. Perhaps due to a change in the concept?
- Journal of Cell Science -- All of their messages are failing, because they're sending HTML with an embedded logo. In order to process the emails, they need to be plain text.
- Palaeontology -- Had some errors, but has been working recently. Due to changes in concept?
- Ecological Applications -- Journal ID in the email is not matching Journal ID in the concept. I'm still working on this one.

**Comment by ryscher** [comment]
Ecological Applications is fixed now -- I had to completely delete the secondary concept.

**Comment by dfagandryad** [comment]
Both Development and Palaeontology had duplicate concepts. I deleted one of the duplicate concepts for each journal on February 10. Getting rid of the duplicates did not seem to resolve the issue immediately, but it is working now.

While reprocessing the email notifications in journal-submit-error, I ran across two additional journals with notifications that do not process correctly. Both have duplicate concepts (shown below), which I am leaving "as-is" for now. I'm not sure why any of the duplicate concepts were created. The newer concepts are only candidates that have not been accepted. Nevertheless, notification emails are getting processing errors. 

The bottom line is there seems to be two issues that may cause ongoing problems if not addressed: (1) Why did the system create duplicate concepts for already existing ones? (2) Why is it causing problems to have duplicate concepts when one of them is only a candidate?

Here are the two aforementioned journals with issues:

- Deutsche Entomologische Zeitschrift (Journal Code: DEZ)
- Ecological Monographs (Journal code:  ecoMono)

**Comment by ryscher** [comment]
When there are duplicate concepts, the way concepts are loaded into the journal module means that the original concept gets pushed out. 

I've deleted the duplicates for the two above journals and reset the journal module so they won't cause problems any more.

I'm not sure why so many duplicate concepts were created; most of them were created about a week before we launched the new system. Regardless, no new concepts are being created automatically, so this will not be an ongoing problem.
